plugins/rapture/nexus-rapture-plugin/src/main/resources/static/rapture/NX/controller/Permissions.js

```diff
@@ -59,12 +60,30 @@ Ext.define('NX.controller.Permissions', {
     );
   },
 
+  /**
+   * Prime initial set of permissions from state.
+   *
+   * @override
+   */
   onLaunch: function () {
-    var me = this;
+    var me = this,
+        rawData = NX.State.getValue('permissions');
+
+    //<if debug>
+    me.logDebug('Initial permissions: ', rawData);
+    //</if>
 
-    me.fetchPermissions();
+    me.getStore('Permission').loadRawData(rawData, false);
+    NX.Permissions.setPermissions(me.getPermissions());
```

Comment:
I can test again, I wasn't seeing the event get fired, or it was lost in the async logging going on at the time, i'm not sure.